apis/management.cattle.io/v3/schema/schema.go

```diff
@@ -183,7 +191,24 @@ func machineTypes(schemas *types.Schemas) *types.Schemas {
 				Output: "machineDriver",
 			}
 		}).
-		MustImport(&Version, v3.MachineTemplate{})
+		MustImport(&Version, v3.MachineTemplate{}).
+		MustImportAndCustomize(&Version, v3.Machine{}, func(schema *types.Schema) {
+			schema.MustCustomizeField("requestedHostname", func(field types.Field) types.Field {
```

Comment:
For types that we define (not a k8s native type) we can just annotate the struct itself and not modify the schema "after the fact."  To do this for "requestedHostname" it would be in the Cluster struct

```go
type Cluster struct [
    RequestedHostname string `json:"requestedHostname,omitempty" norman:"type=dnsLabel,nullable,required"`
}
```

The approach does not work for the `name` field because we do define that field and it comes from ObjectMeta.  So for "name" you must use ImportAndCustomize approach